src/vs/workbench/contrib/files/browser/editors/binaryFileEditor.ts

```diff
@@ -49,15 +49,29 @@ export class BinaryFileEditor extends BaseBinaryResourceEditor {
 			// Try to let the user pick an override if there is one availabe
 			const overridenInput = await this.editorOverrideService.resolveEditorOverride(input, { ...options, override: EditorOverride.PICK, }, this.group);
 
-			// Replace the overrriden input, with the text based input
-			await this.editorService.replaceEditors([{
-				editor: input,
-				replacement: overridenInput?.editor ?? input,
-				options: {
-					...overridenInput?.options ?? options,
-					override: EditorOverride.DISABLED
-				}
-			}], overridenInput?.group ?? this.group);
+			if (overridenInput === OverrideStatus.ABORT) {
+				return;
+			} else if (overridenInput === OverrideStatus.NONE) {
+				// Replace the overrriden input, with the text based input
```

Comment:
There is some repetition between this block and the next one. Can we extract the reusable thing (e.g. the call to `replaceEditors`) and only massage the input as needed?